### PR TITLE
Route for product location association

### LINKED ISSUE #46: Route for product - location association
We need a route for product - location association (the same way we did it for the reviews)

### PR DESCRIPTION
**Pull request by matthewlanders**
Closes #46

### CODE REVIEW

File: server/app/routes/orders/index.js [hunk 1]
```diff
@@ -20,6 +21,16 @@ router.param('id', function(req, res, next, theId){
     .catch(next);
 });
 
+router.param('uId', function(req, res, next, theId){
```
Comment (coreygreenwald) [inline server/app/routes/orders/index.js]: I think req.user.id (and req.user) takes care of this.

File: server/app/routes/products/index.js [hunk 2]
```diff
@@ -29,9 +30,23 @@ router.get('/', function (req, res, next) {
...
 
 router.post('/', function(req, res, next){
-	Product.create(req.body)
+	var myProduct;
+	Product.create(req.body.productData)
 	.then(function(createdProduct){
-		res.status(201).send(createdProduct);
+		myProduct = createdProduct;
+	})
+	.then(function(){
```
Comment (coreygreenwald) [inline server/app/routes/products/index.js]: don't need this .then

File: server/app/routes/products/index.js [hunk 3]
```diff
@@ -29,9 +30,23 @@ router.get('/', function (req, res, next) {
...
+			where:{
+				name: req.body.location
+			}
+		})
+	})
+	.then(function(foundLocation){
+		return myProduct.setLocation(foundLocation)
+	})
+	.then(function(updatedProduct){
+		res.status(201).send(updatedProduct);
```
Comment (coreygreenwald) [inline server/app/routes/products/index.js]: The rest looks good.

File: tests/server/routes/orders-test.js [hunk 4]
```diff
@@ -98,15 +99,32 @@ describe('Orders Route', function () {
...
     describe('POST one order', function(done){
 
+        var user;
+
+        beforeEach('Create a user', function (done) {
+            return User.create({
+                    firstName: 'Matt',
+                    lastName : 'Landers',
+                    email : 'mattlanders@smartpeople.com',
+                    password : 'Jennaisthebestandsmartest'
```
Comment (coreygreen1108) [inline tests/server/routes/orders-test.js]: suck up 😛

**Pull request merged**